File: TypeScript/Object Validator/src/runtime/rule.ts
import { Rule as ConfigRule } from "../config";
import Condition from "./condition";
import { buildRuntimeCondition } from "./utils";

export default class Rule {
    readonly condition: Condition;
    readonly id?: number;
    readonly errorMessage?: string;

    constructor(rule: ConfigRule) {
        this.condition = buildRuntimeCondition(rule.condition);
        this.id = rule.id;
        this.errorMessage = rule.errorMessage;
    }
}

File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/Rule.java
package com.quicksilver.objectvalidator.runtime;

public class Rule {
    public final Condition condition;
    public final Integer id;
    public final String errorMessage;

    public Rule(com.quicksilver.objectvalidator.config.Rule rule) {
        this.condition = Utils.buildRuntimeCondition(rule.condition);
        this.id = rule.id;
        this.errorMessage = rule.errorMessage;
    }
}
